### ISSUE TITLE
EarlyStopping Callback not working with Multiple Callbacks

### ISSUE THREAD

**Issue by eduardofv** [tensorflow/tensorflow#37587]
**System information** 
- Have I written custom code  
- OS Platform and Distribution: *Google Colab*
- TensorFlow version: *2.1.0*
- Keras: *2.2.4-tf*
- Python version: *3* 
- CUDA/cuDNN version: No GPU but happens also with it

**Describe the current behavior**
When a CustomCallback  uses `model.predict` or `model.evaluate` as part of `on_epoch_end`,  to for example, track a custom evaluation metric on an test dataset, `model.stop_training` is reset to False even if it was previously set to True by Early Stopping. Training does not stop. Current workaround is putting EarlyStopping as the last callback of the list.

**Describe the expected behavior**
Regardless the order in which callbacks are called, if one of the sets the model to stop training it should stop even if a later callback resets the stop flag.

**Standalone code to reproduce the issue** 
Check this colab with the full example: https://colab.research.google.com/drive/1lw943Ggwkp_wvGxVX-5XqaEJ5qXmrHlA

In short:
```
class MyCallback(keras.callbacks.Callback):
    def __init__(self, test_data):
        super(MyCallback, self).__init__()
        self.test_data = test_data

    def on_epoch_end(self, epoch, logs=None):
        print(f"\n--------- pre-predict stop_training={self.model.stop_training}\n")
        #The problem is in the prediction: if commented ES works fine
        predictions = self.model.predict(self.test_data.batch(512))
        print(f"\n--------- post-predict stop_training={self.model.stop_training}\n")
```
```
es = keras.callbacks.EarlyStopping(patience=2)
myc = MyCallback(test_data)

#This causes EarlyStop not to stop
my_callbacks = [es, myc]
#Either of these works fine
#my_callbacks = [myc, es]
#my_callbacks = [es]
...
model.fit(train_data.batch(512),
          validation_data=validation_data.batch(512),
          epochs=100,
          callbacks=my_callbacks,
          verbose=1)
```

This issue is refered also in the keras repository: https://github.com/keras-team/keras/issues/13381

Some thoughts: I'm not sure if the correct solution would be to discourage the use of predict or evaluate inside a training loop since there may be some other side effects of running one of those to the model.

Anyway, I'm opening the issue and submitting a pull request to fix this.

**Comment by gadagashwini** [comment]
I am able to reproduce the issue with Tf 2.1.
Please find the gist [here](https://colab.sandbox.google.com/gist/gadagashwini/756228eeeca347ffa8f4a40cd662d7f1/untitled457.ipynb). Thanks!

**Comment by eduardofv** [comment]
As suggested on the [pull request](https://github.com/tensorflow/tensorflow/pull/37588) it's now fixed on v2.2rc0 and current master. Closing the issue. Thanks!
